python/cudf/cudf/core/frame.py

```diff
@@ -65,7 +65,12 @@ def __init_subclass__(cls):
 
     @classmethod
     def _from_table(cls, table: Frame):
-        return cls(table._data, index=table._index)
+        return cls(
```

Comment:
I'm not sure why this change is in this PR right now since you're not using `_from_table` anywhere that I can see. In case you do need it, though, please use `_from_data` instead (see #8687, `_from_table` is being removed).

Reply:
It ends up being used inside `_apply_boolean_mask`. The problem ends up being that what comes out of `libcudf.stream_compaction.apply_boolean_mask` is a `Table` whose index is also a `Table`. Without this change it never gets converted to an actual index, and so it has no `_copy_type_metadata` function and we crash:

```
AttributeError: 'cudf._lib.table.Table' object has no attribute '_copy_type_metadata'
```

Since there's already a change in flight to remove this function entirely I decided not to try and rework that piece of things in this PR and just fix the bug in the current context, however short lived that might be.

Reply:
Ah I see this is a stopgap until that PR is merged. Sounds good.